spec/services/workflow_errors_reporter_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'rails_helper'
+
+RSpec.describe WorkflowErrorsReporter do
+  let(:put) { 'https://sul-lyberservices-test.stanford.edu/workflow/dor/objects/druid:jj925bx9565/workflows/preservationWF/moab-valid' }
+  let(:headers) { { 'Content-Type' => 'application/xml' } }
+
+  describe '.update_workflow' do
+    let(:error_msg) do
+      { 13 => "PreservedObjectHandler(jj925bx9565, 9, 6570668, <Endpoint:...>) Invalid moab, validation error...ential version directories. Current directories: [\\\"v0001\\\", \\\"v0002\\\", \\\"v0009\\\"]\"]" }
+    end
+
+    let(:body) { "<process name='moab-valid' status='error' errorMessage='#{error_msg}'/>" }
+
+    before do
+      Settings.workflow_services_url = 'https://sul-lyberservices-test.stanford.edu/workflow/'
+      stub_request(:put, put)
+        .with(body: body,
+              headers: headers)
+        .to_return(status: 204, body: "", headers: {})
+    end
+    it 'updates preservationWF with invalid moab errors' do
+      expect(described_class.update_workflow('jj925bx9565', 'moab-valid', error_msg)).to be_an_instance_of(Faraday::Response)
+    end
+
+    it 'rescues for Faraday::Error' do
+      allow(described_class).to receive(:update_workflow).and_raise(Faraday::Error)
+      expect { described_class.update_workflow('jj925bx9565', 'moab-valid', 'Foo() error') }.to raise_error(Faraday::Error)
```

Comment:
Trying to find examples for you (not even sure you need webmock?):

- https://github.com/sul-dlss/sul-requests/blob/master/spec/models/library_hours_api_spec.rb#L18-L25
- https://github.com/sul-dlss/sul-requests/blob/master/spec/models/symphony_curr_loc_request_spec.rb#L48-L64
- https://github.com/sul-dlss/sul-requests/blob/master/spec/models/symphony_user_name_request_spec.rb#L27-L51
- https://github.com/sul-dlss/sul-requests/blob/master/spec/models/proxy_access_spec.rb#L64-L98

Reply:
- https://github.com/sul-dlss/argo/blob/master/spec/models/index_queue_spec.rb#L7-L12

- https://github.com/sul-dlss/dor-workflow-service/blob/master/spec/workflow_service_spec.rb#L76-L79 <-- might be the closest to the sort of test I *think* you were going for???

Reply:
Note also:  do we want to test the response code from Dor workflows?  Make sure it didn't get an error?  Log it if it did?

There can be a connection failure;  there can be bad http response codes ...

Reply:
okay cool - i'll try to disable webmock and try these